ros_buildfarm/workspace.py

```diff
@@ -116,7 +116,9 @@ def call_build_tool(
     for parent_result_space in reversed(parent_result_spaces):
         setup_file = os.path.join(parent_result_space, 'setup.sh')
         if os.path.exists(setup_file):
-            cmd = '. %s && %s' % (setup_file, cmd)
+            setup_cmd = 'COLCON_CURRENT_PREFIX=%s _CATKIN_SETUP_DIR=%s . %s' % (
```

Comment:
By introspecting the content of the `parent_result_space` directory it should be possible to decide which of the two environment variables is needed (and in case of `colcon` even if it is needed at all):
* contains a `.catkin` file: `_CATKIN_SETUP_DIR`
* contains a `.colcon_install_layout` file: `COLCON_CURRENT_PREFIX`
* contains neither (this should be ROS 2 Debian packages): *no variable is needed*